### PR TITLE
DOCS: Formatting and phrasing updates for SHAP return values

### PR DESCRIPTION
**Pull request by connortann**
## Overview

A few fixups following #3318

- Update RST formatting for `versionchanged` directive and bullet points
- Rephrase docstrings slightly for clarity (feedback/suggestions welcome)

### CODE REVIEW

File: shap/explainers/_gradient.py [hunk 1]
```diff
@@ -131,19 +131,26 @@ def shap_values(self, X, nsamples=200, ranked_outputs=None, output_rank_order="m
...
-              - multiple inputs one or more outputs: list of matrices, with shapes of one of the above.
-            If ranked_outputs is None then this list of tensors matches
-            the number of model outputs. If ranked_outputs is a positive integer a pair is returned
-            (shap_values, indexes), where shap_values is a list of tensors with a length of
-            ranked_outputs, and indexes is a matrix that tells for each sample which output indexes
-            were chosen as "top".
-
-           .. versionchanged:: 0.45.0
-           Return type for models with multiple outputs and one input changed from list to np.ndarray.
+        np.array or list
```
Comment (CloseChoice) [inline shap/explainers/_gradient.py:134]: We still have just `array` for the deepexplainer. I am supportive of `np.array` cause it is clearer. Could you adapt that in all the explainers, please?

**Review by CloseChoice — CHANGES_REQUESTED** [review]
LGTM.

**Review by CloseChoice — APPROVED** [review]
LGTM, sorry I confused this with another PR.

File: shap/explainers/_tree.py [hunk 2]
```diff
@@ -396,15 +396,20 @@ def shap_values(self, X, y=None, tree_limit=None, approximate=False, check_addit
         Returns
         -------
         array
```
Comment (CloseChoice) [inline shap/explainers/_tree.py]:
```suggestion
        np.array
```

**Review by CloseChoice — APPROVED** [review]
One more optional change, otherwise LGTM

File: shap/explainers/_gradient.py (hunk 1 above)
Comment (connortann) [reply shap/explainers/_gradient.py:134]: Done

**Pull request merged**